test/e2e-common.sh

```diff
@@ -210,6 +211,78 @@ function enable_monitoring(){
       --role roles/cloudtrace.agent
 }
 
+function test_authentication_check_for_brokercell() {
+  echo "Starting authentication check test for brokercell."
+  local auth_mode=${1}
+  ko apply -f ${CONFIG_WARMUP_GCP_BROKER}
+
+  echo "Starting authentication check test which is running outside of the broker related Pods."
+  wait_until_brokercell_authentication_check_pending "$(non_pod_check_keywords)" || return 1
+
+  echo "Starting authentication check test which is running inside of the broker related Pods."
+  apply_invalid_resource "$auth_mode"
+  wait_until_brokercell_authentication_check_pending "$(pod_check_keywords "$auth_mode")" || return 1
+
+  # Clean up all the testing resources.
+  echo "Authentication check test finished, waiting until all broker related testing resources deleted."
+  delete_invalid_resource "$auth_mode"
+  ko delete -f ${CONFIG_WARMUP_GCP_BROKER}
+  kubectl delete brokercell default -n "${CONTROL_PLANE_NAMESPACE}"
+  kubectl wait pod --for=delete -n "${CONTROL_PLANE_NAMESPACE}" --selector=brokerCell=default --timeout=5m || return 1
+}
+
+function wait_until_brokercell_authentication_check_pending() {
+  local keywords=${1}
+  echo "Waiting until brokercell authentication check pending."
+  for i in {1..150}; do #timeout after 5 minutes
+    local reason=$(kubectl get brokercell default -n "${CONTROL_PLANE_NAMESPACE}" -o=jsonpath="{range .status.conditions[*]}{.reason}" --ignore-not-found=true)
+    local message=$(kubectl get brokercell default -n "${CONTROL_PLANE_NAMESPACE}" -o=jsonpath="{range .status.conditions[*]}{.message}" --ignore-not-found=true)
```

Comment:
Remove `reason` and use this, which I think provides the same output as the current `message` line.

```suggestion
    local message=$(kubectl get brokercell default -n "${CONTROL_PLANE_NAMESPACE}" -o=jsonpath='{.status.conditions[?(@.reason == "AuthenticationCheckPending")].message}' --ignore-not-found=true)
```